env0/resource_project.go

```diff
@@ -181,17 +174,16 @@ func resourceProjectDelete(ctx context.Context, d *schema.ResourceData, meta int
 				case <-ticker.C:
 					err := resourceProjectAssertCanDelete(ctx, d, meta)
 
-					if err == nil {
-						done <- true
-						return
-					}
-
-					if aeerr, ok := err.(*ActiveEnvironmentError); ok {
-						if !aeerr.retry {
-							done <- true
-							return
+					if err != nil {
```

Comment:
tests were failing. This is now a more simplified version.
If error and error is retry, then retry.
In all other cases we're done (error or no error).